### PR TITLE
fix kubeflow pipeline

### PR DESCRIPTION
**Pull request by J-shang**

### CODE REVIEW

File: ts/nni_manager/training_service/kubernetes/kubeflow/kubeflowTrainingService.ts [hunk 1]
```diff
@@ -310,7 +312,7 @@ class KubeflowTrainingService extends KubernetesTrainingService implements Kuber
         // Generate kubeflow job resource config object
         const kubeflowJobConfig: any = await this.generateKubeflowJobConfig(trialJobId, trialWorkingFolder, kubeflowJobName, workerPodResources,
                                                                       nonWorkerResources);
-
+        this.log.info(`${JSON.stringify(kubeflowJobConfig)}`)
```
Comment (cruiseliu) [inline ts/nni_manager/training_service/kubernetes/kubeflow/kubeflowTrainingService.ts:315]: Don't use `JSON.stringify`. Logger will format it. `JSON.stringify` could crash NNI manager if the content contains non-JSON values.

File: ts/nni_manager/training_service/reusable/environments/kubernetes/kubeflowEnvironmentService.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ export class KubeflowEnvironmentService extends KubernetesEnvironmentService {
         // Create kubernetesCRDClient
         this.kubernetesCRDClient = KubeflowOperatorClientFactory.createClient(
             this.config.operator, this.config.apiVersion);
+        this.kubernetesCRDClient.setNamespace = this.config.namespace ? this.config.namespace : "default";
```
Comment (cruiseliu) [inline ts/nni_manager/training_service/reusable/environments/kubernetes/kubeflowEnvironmentService.ts:29]: The JS way: `this.config.namespace ?? 'default'`

File: ts/nni_manager/training_service/kubernetes/kubernetesApiClient.ts [hunk 3]
```diff
@@ -158,6 +159,14 @@ abstract class KubernetesCRDClient {
 
     protected abstract get operator(): any;
 
+    public set setNamespace(namespace: string) {
```
Comment (cruiseliu) [inline ts/nni_manager/training_service/kubernetes/kubernetesApiClient.ts:162]: Seem it's not needed and this is not how setter works. It's like following code in Python:
```python
@property
def getNamespace(self):
    return self.namespace

@setNamespace.setter
def setNamespace(self, namespace):
    self.namespace = namespace
```

File: ts/nni_manager/training_service/reusable/environments/kubernetes/kubeflowEnvironmentService.ts (hunk 2 above)
Comment (J-shang) [reply ts/nni_manager/training_service/reusable/environments/kubernetes/kubeflowEnvironmentService.ts:29]: got it

File: ts/nni_manager/training_service/kubernetes/kubernetesApiClient.ts (hunk 3 above)
Comment (J-shang) [reply ts/nni_manager/training_service/kubernetes/kubernetesApiClient.ts:162]: just follow code style in `GeneralK8sClient `...

File: test/config/examples/cifar10_search_space.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
     "lr":{"_type":"choice", "_value":[0.1, 0.01, 0.001, 0.0001]},
     "optimizer":{"_type":"choice", "_value":["SGD", "Adadelta", "Adagrad", "Adam", "Adamax"]},
-    "model":{"_type":"choice", "_value":["vgg", "resnet18"]}
+    "model":{"_type":"choice", "_value":["vgg"]}
```
Comment (liuzhe-lz) [inline test/config/examples/cifar10_search_space.json:4]: Single candidate choice looks dangerous to me...

File: nni/experiment/config/training_services/kubeflow.py [hunk 5]
```diff
@@ -43,6 +43,7 @@ class KubeflowConfig(TrainingServiceConfig):
     ps: Optional[KubeflowRoleConfig] = None
     master: Optional[KubeflowRoleConfig] = None
     reuse_mode: Optional[bool] = True #set reuse mode as true for v2 config
+    namespace: Optional[str] = 'default'
```
Comment (liuzhe-lz) [inline nni/experiment/config/training_services/kubeflow.py:46]: If `null` has same effect with `'default'`, the type should be `str`.

File: ts/nni_manager/common/experimentConfig.ts [hunk 6]
```diff
@@ -132,6 +132,7 @@ export interface KubeflowConfig extends TrainingServiceConfig {
     master?: KubeflowRoleConfig;
     reuseMode: boolean;
     maxTrialNumberPerGpu?: number;
+    namespace?: 'default';
```
Comment (liuzhe-lz) [inline ts/nni_manager/common/experimentConfig.ts:135]: The type is `string`. `'default'` means `Literal['default']` in python.

File: ts/nni_manager/config/aml/amlUtil.py [hunk 7]
```diff
@@ -52,7 +52,7 @@
                     print('stop_result:failed')
                     exit(0)
                 loop_count += 1
-                time.sleep(500)
+                time.sleep(5)
```
Comment (liuzhe-lz) [inline ts/nni_manager/config/aml/amlUtil.py:55]: `status` is never updated in loop body, so the result is deterministic............

File: ts/nni_manager/training_service/kubernetes/kubeflow/kubeflowTrainingService.ts [hunk 8]
```diff
@@ -310,7 +312,7 @@ class KubeflowTrainingService extends KubernetesTrainingService implements Kuber
         // Generate kubeflow job resource config object
         const kubeflowJobConfig: any = await this.generateKubeflowJobConfig(trialJobId, trialWorkingFolder, kubeflowJobName, workerPodResources,
                                                                       nonWorkerResources);
-
+        this.log.info(`${kubeflowJobConfig}`)
```
Comment (liuzhe-lz) [inline ts/nni_manager/training_service/kubernetes/kubeflow/kubeflowTrainingService.ts:315]: No string formatting...
Just `this.log.info('kubeflowJobConfig:', kubeflowJobConfig);`

File: ts/nni_manager/training_service/kubernetes/kubernetesApiClient.ts (hunk 3 above)
Comment (liuzhe-lz) [reply ts/nni_manager/training_service/kubernetes/kubernetesApiClient.ts:162]: If you want to use property, it should be:
```typescript
public get namespace(): string {
    return this.privateNamespace;
}
public set namespace(ns: string): void {
    this.privateNamespace = ns;
}
```
Or if you want to use getter method, it should be:
```typescript
public getNamespace(): string {
    return this.namespace;
}
public setNamespace(namespace: string): void {
    this.namespace = namespace;
}
```

Comment (liuzhe-lz) [reply ts/nni_manager/training_service/kubernetes/kubernetesApiClient.ts:162]: I suggest to simply make the field public...

File: ts/nni_manager/config/aml/amlUtil.py (hunk 7 above)
Comment (liuzhe-lz) [reply ts/nni_manager/config/aml/amlUtil.py:55]: Oh it's in next line. My mistake.

**Review by liuzhe-lz — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]

**Pull request merged**